instagram/src/App.js

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from "react";
+import dummyData from "./dummy-data";
+import SearchBar from "./components/SearchBar/SearchBar";
+import PostContainer from "./components/PostContainer/PostContainer";
+import "./App.css";
+
+class App extends Component {
+  state = {
+    posts: dummyData,
+  };
+
+  handleAddComment = (comment, post) => {
```

Comment:
This is great! What a flawless execution of this task. If you didn't have to pass the post in, however it might be easier to just add the comment to the comment array found on the CommentSection's state.

There is a bit of a problem here but it's only a slight problem. While passing that post around, you're passing it as a referenced object, which is why your code on line 16 returns true. 

Here's the lesson:

```
const myObj = {name: 'ryan'};
const myObj2 = {name: 'ryan'};
myObj === myObj2  -> returns false. 
These are separate objects not sharing space in memory.
```
example 2 (what you have going on that is a bit problematic);
```
const myObj = {name: 'Lauren'};
const myObj2 = myObj;
myObj === myObj2 -> returns true
These objects share space in memory.
```

So, you're passing around the entire object when this function is invoked (which is totally fine) but then you _could_ potentially create side-effects although you're using the spread operator and essentially `Object.Assign()` and then concatenating the object's array on comments together with the newly added comment... wheeew.. lots to explain and I'm sure you already knew all of this. 

The idea here, and the idea that i gave in the lecture, was to simplify this algorithm by separating our comments away from the actual post object by passing them down as props and setting them on the CommentSection's state.. why does this simply our lives? Well now our algorithm is as simple as.

```
1. build out comment object w/ username, comment and id,
2. clone the state array
3. push comment into clone
4. call setState
```

Quite a few extra steps, however I'm thoroughly impressed with your algorithm.. there are just simply things to consider when it comes to ... simplicity... 

Great stuff